lib/foreman/renderer.rb

```diff
@@ -28,8 +32,9 @@ def get_source(klass: Foreman::Renderer::Source::Database, template:, **args)
         klass.new(template)
       end
 
-      def get_scope(klass: Foreman::Renderer::Scope::Provisioning, host: nil, params: {}, variables: {})
-        klass.new(host: host, params: params, variables: variables)
+      def get_scope(klass: nil, host: nil, params: {}, variables: {}, mode: REAL_MODE, template: nil)
```

Comment:
Maybe we can pass `klass` here, instead of `template` instance?

Reply:
sorry, not sure I follow, if we pass klass as an argument, it wins `klass ||= `, then we default to template's default class and then we fallback to Foreman::Renderer::Scope::Provisioning